### PR TITLE
Switch build backend to Hatchling

### PR DESCRIPTION
**Pull request by ofek**
cc @hugovk 

Hatchling is the minimal build backend of [Hatch](https://github.com/ofek/hatch) (v1 rewrite of CLI being released soon). You can think of it as the same distinction between `flit-core` & `flit` or `poetry-core` & `poetry`, with `hatchling` being more similar to `flit-core` in that only accepted standards are used e.g. project metadata format ([PEP 621](https://www.python.org/dev/peps/pep-0621/)), dependency specification ([PEP 631](https://www.python.org/dev/peps/pep-0631/)), etc.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -330,160 +330,3 @@ Credits
   `PyPI`_.
 - `Paul Kehrer <https://github.com/reaperhulk>`_ for his
   `awesome blog post <https://langui.sh/2016/12/09/data-driven-decisions>`_.
-
-Changelog
```
Comment (hugovk) [inline README.rst:334]: Shall we add a link to the new changeling file? It could be a plain link or a badge.

**Review by hugovk — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (ofek) [reply README.rst:334]: I added:

```toml
[project.urls]
Changelog = "https://github.com/ofek/pypinfo/blob/master/CHANGELOG.rst"
```

which should add a button on https://pypi.org/project/pypinfo/. Should I also add a link here?

Comment (hugovk) [reply README.rst:334]: I saw, that's really good :)

That should be fine, we can always add it later.

**Comment by hugovk** [comment]
@ofek I just spotted after merging this, the deploy workflow failed on `python setup.py --version`, because of course that file no longer exists.

https://github.com/ofek/pypinfo/runs/4825441962?check_suite_focus=true

Is there a similar way to print out the version with this new method? If not, it's not an essential step and we can simply remove it.

**Comment by ofek** [comment]
No standard way. Hatch's new CLI can but that would require installation

**Pull request merged**